packages/client/src/components/app/dynamic-filter/DynamicFilter.svelte

```diff
@@ -61,8 +61,9 @@
       clonedSchema = schema
     } else {
       allowedFields?.forEach(field => {
-        if (schema[field]) {
-          clonedSchema[field] = schema[field]
+        if (schema[field.name]) {
```

Comment:
As we have changed the type of this setting from an array of strings to an array of objects, we need to ensure that the `DynamicFilter` component handles both. The value of the setting is only migrated whenever the setting component is rendered, so there is no guarantee that all instances are migrated.

We need to update this logic to account for both flat arrays of strings (old structure) and arrays of objects (new structure).